File: request/request.java

      
       
    import java.io.BufferedReader;
    import java.io.InputStreamReader;
    import java.net.Authenticator;
    import java.net.HttpURLConnection;
    import java.net.InetSocketAddress;
    import java.net.PasswordAuthentication;
    import java.net.Proxy;
    import java.net.URL;

    public class Socks5ProxyRequest {

        public static void main(String[] args) {
            String proxyHost = "ip"; // Proxy host
            int proxyPort = port; // Proxy port
            String proxyUser = "your_username"; // Proxy username (replace with your proxy username, e.g., USERxxxx-zone-custom)
            String proxyPass = "your_password"; // Proxy password (replace with your proxy password)
            // Target URL
            String targetUrl = "https://ipinfo.io/json/";

            try {

                Authenticator.setDefault(new Authenticator() {
                    @Override
                    protected PasswordAuthentication getPasswordAuthentication() {
                        return new PasswordAuthentication(proxyUser, proxyPass.toCharArray());
                    }
                });

                // Create proxy
                Proxy proxy = new Proxy(Proxy.Type.SOCKS, new InetSocketAddress(proxyHost, proxyPort));

                // Create URL object
                URL url = new URL(targetUrl);

                HttpURLConnection connection = (HttpURLConnection) url.openConnection(proxy);

                // Set Request
                connection.setRequestMethod("GET");
                connection.setConnectTimeout(10000);
                connection.setReadTimeout(10000);
                connection.setRequestProperty("User-Agent", "Mozilla/5.0");

                //Connect and get response code
                int responseCode = connection.getResponseCode();

                if (responseCode == 200) {
                    // Read response
                    BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
                    String inputLine;
                    StringBuilder response = new StringBuilder();

                    while ((inputLine = in.readLine()) != null) {
                        response.append(inputLine);
                    }
                    in.close();

                    // Print
                    System.out.println(response.toString());
                } else {
                    System.out.println(responseCode);
                }

                // Disconnect
                connection.disconnect();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

          
    

File: request/request.php

      
       
    <?php
    // SOCKS5 proxy configuration
    $proxyHost = 'ip'; // Proxy host
    $proxyPort = '10000'; // Proxy port
    $proxyUser = 'your_username'; // Proxy username (replace with your proxy username, e.g., USERxxxx-zone-custom)
    $proxyPass = 'your_password'; // Proxy password (replace with your proxy password)
    // IP info URL
    $ipInfoUrl = 'https://ipinfo.io/json';

    // Initialize cURL
    $ch = curl_init();

    // Set cURL options
    curl_setopt($ch, CURLOPT_URL, $ipInfoUrl); // URL to request
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); // Return the response as a string
    curl_setopt($ch, CURLOPT_PROXY, $proxyHost . ':' . $proxyPort); // Set the SOCKS5 proxy
    curl_setopt($ch, CURLOPT_PROXYTYPE, CURLPROXY_SOCKS5_HOSTNAME); // Set proxy type as SOCKS5 and resolve DNS through proxy
    curl_setopt($ch, CURLOPT_PROXYUSERPWD, $proxyUser . ':' . $proxyPass); // Proxy username and password
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false); // Disable SSL peer verification (for testing only)
    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false); // Disable SSL host verification (for testing only)
    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 30); // 30-second connection timeout
    curl_setopt($ch, CURLOPT_TIMEOUT, 60); // 60-second total timeout

    // Execute the request
    $response = curl_exec($ch);

    // Check for cURL errors
    if ($response === false) {
        echo "cURL Error: " . curl_error($ch) . "";
    } else {
        echo "Response from IP info:";
        echo $response;
    }

    // Close the cURL session
    curl_close($ch);

          
    
